### PR TITLE
Python client: Plumb ca_bundle to requests

### PR DESCRIPTION
**Pull request by eranmes**
In order to verify the SSL certificate received for https requests, the
ca_bundle has to be passed into the requests module. Needed when the
ca_bundle is not in the standard place.

### CODE REVIEW

File: python/ct/client/log_client.py [hunk 1]
```diff
@@ -196,8 +196,9 @@ def _parse_consistency_proof(response, servername):
 # A class that we can mock out to generate fake responses.
 class RequestHandler(object):
     """HTTPS requests."""
-    def __init__(self, connection_timeout=60):
+    def __init__(self, connection_timeout=60, ca_bundle=True):
```
Comment (phad) [inline python/ct/client/log_client.py]: ca_bundle doesn't sound like something you'd pass as a boolean value.
is it really a flag?

Comment (eranmes) [inline python/ct/client/log_client.py]: The ca_bundle is being passed into verify, which can take one of the following values:
True: SSL certificate validation will take place using the default CA root certs set.
False: SSL certificate validation will not take place.
Path to a file: SSL certificate validation will take place using the CA root set specified in this file.

See requests documentation:
http://docs.python-requests.org/en/latest/user/advanced/#ssl-cert-verification

Comment (phad) [inline python/ct/client/log_client.py]: Yeuch.  An argument that can be given a boolean value, or string.

**Comment by benlaurie** [comment]
Hmm ... given the weirdness (boolean vs. string):

a) Documentation?

b) Should we decline a false setting? We should not encourage disabling cert checking!

**Pull request merged**